cpp/tests/io/compression_common.hpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/io/types.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <list>
+#include <string>
+#include <tuple>
+
+class tmp_env_var {
+ public:
+  explicit tmp_env_var(std::string const& name, std::string const& value) : name_(name)
```

Comment:
For my own _edumacation_:  Should constructor variables whose values are adopted be passed by value?
```suggestion
  explicit tmp_env_var(std::string name, std::string const& value) : name_{std::move(name)}
```
I realize that it breaks symmetry with `value`.  Nor is this a perf path.  I'm mostly just curious about the convention, so please feel free to ignore.

Reply:
yeah, should be by value + move. I keep defaulting to the ref + copy pattern out of habit.